mobile/src/reducers/mobile.js

```diff
@@ -1,25 +1,35 @@
-import { SETUP, SET_URL, SET_STATE, ERROR } from '../actions'
+import {
+  SETUP, SET_URL, SET_STATE, ERROR, UPDATE_SETTINGS,
+  SHOW_UNLINK_CONFIRMATION, HIDE_UNLINK_CONFIRMATION, UNLINK
+} from '../actions'
 
 const initialState = {
   isSetup: false,
   serverUrl: '',
   settings: {
-    backupImages: false
+    backupImages: false,
+    displayUnlinkConfirmation: false
   }
 }
 
 export const mobile = (state = initialState, action) => {
   switch (action.type) {
     case SET_STATE:
-      return action.state
+      return Object.assign({}, initialState, action.state)
```

Comment:
if default value of `state` is `initialState`, we may stay pure and use `state` instead of `initialState`:

```javascript
return Object.assign({}, state, action.state)
```

but as soon as we write this, we understand that the action is not correct:

- SET_STATE should set **a** state
- UPDATE_STATE should update **the** state
- INITIALIZE_STATE should initialize **the** state
- and so on...

We may create another case to match the wanted behavior.